src/coreclr/jit/target.h

```diff
@@ -208,21 +215,186 @@ enum _regMask_enum : unsigned
 // In any case, we believe that is OK to freely cast between these types; no information will
 // be lost.
 
+typedef unsigned __int32 RegBitSet32;
+
 #if defined(TARGET_AMD64) || defined(TARGET_ARMARCH) || defined(TARGET_LOONGARCH64) || defined(TARGET_RISCV64)
 typedef unsigned __int64 regMaskTP;
+typedef unsigned __int64 regMaskGpr;
+typedef unsigned __int64 regMaskFloat;
+typedef unsigned __int64 regMaskPredicate;
+typedef unsigned __int64 RegBitSet64;
+
+// Represents that the mask in this type is from one of the register type - gpr/float/predicate
+// but not more than 1.
+typedef unsigned __int64 regMaskOnlyOne;
```

Comment:
Is it possible to have this type come with a tag that can ensure (in DEBUG only) that we don't try to do operations with mismatched register types? Currently I assume that would result in bogus results.